kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/GCPAuthenticatorUtils.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.utils;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import io.fabric8.kubernetes.api.model.NamedContext;
+import io.fabric8.kubernetes.client.internal.KubeConfigUtils;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class for GCP token refresh.
+ */
+public class GCPAuthenticatorUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GCPAuthenticatorUtils.class);
+
+  public static final String EMPTY = "";
+  public static final String ACCESS_TOKEN_PARAM = "access_token";
+  public static final String EXPIRY_PARAM = "expiry";
+  public static final String SCOPES = "scopes";
+  public static final String[] DEFAULT_SCOPES =
+    new String[]{
+      "https://www.googleapis.com/auth/cloud-platform",
+      "https://www.googleapis.com/auth/userinfo.email"
+    };
+  private static GoogleCredentials credentials;
+
+  private GCPAuthenticatorUtils() {
+  }
+
+  /**
+   *  Google Application Credentials-based refresh
+   *  https://cloud.google.com/kubernetes-engine/docs/how-to/api-server-authentication#environments-without-gcloud
+   * @param currentAuthProviderConfig current AuthInfo's AuthProvider config as a map
+   * @return access token for interacting with Google Kubernetes API
+   */
+  public static CompletableFuture<String> resolveTokenFromAuthConfig(
```

Comment:
I don't understand the usage of a CompletableFuture here, both*** return statements return sync values `CompletableFuture.completedFuture`

Also, line 72 seems to be missing a `return`?